ui/app/components/secret-delete-menu.js

```diff
@@ -0,0 +1,147 @@
+    'capabilities',
+    context => {
+      if (!context.args || !context.args.modelForData || !context.args.modelForData.id) return;
+      let [backend, id] = JSON.parse(context.args.modelForData.id);
+      return {
+        id: `${backend}/delete/${id}`,
+      };
+    },
+    'model.id'
+  )
+  deleteVersionPath;
+  @alias('deleteVersionPath.canUpdate') canDeleteAnyVersion;
+
+  @maybeQueryRecord(
+    'capabilities',
+    context => {
+      if (!context.args || !context.args.modelForData || !context.args.modelForData.id) return;
+      let [backend, id] = JSON.parse(context.args.modelForData.id);
+      return {
+        id: `${backend}/undelete/${id}`,
+      };
+    },
+    'model.id'
+  )
+  undeleteVersionPath;
+  @alias('undeleteVersionPath.canUpdate') canUndeleteVersion;
+
+  @maybeQueryRecord(
+    'capabilities',
+    context => {
+      if (!context.args || !context.args.modelForData || !context.args.modelForData.id) return;
+      let [backend, id] = JSON.parse(context.args.modelForData.id);
+      return {
+        id: `${backend}/destroy/${id}`,
+      };
+    },
+    'model.id'
+  )
+  destroyVersionPath;
+  @alias('destroyVersionPath.canUpdate') canDestroyVersion;
+
+  @maybeQueryRecord(
+    'capabilities',
+    context => {
+      if (!context.args.model || !context.args.model.engine || !context.args.model.id) return;
+      let backend = context.args.model.engine.id;
+      let id = context.args.model.id;
+      return {
+        id: `${backend}/metadata/${id}`,
+      };
+    },
+    'model',
+    'model.id',
+    'mode'
+  )
+  v2UpdatePath;
+  @alias('v2UpdatePath.canDelete') canDestroyAllVersions;
+
+  get isLatestVersion() {
+    let { model } = this.args;
+    if (!model) return false;
+    let latestVersion = model.currentVersion;
+    let selectedVersion = model.selectedVersion.version;
+    if (latestVersion !== selectedVersion) {
+      return false;
+    }
+    return true;
+  }
+
+  @action toggleDeleteModal() {
+    this.showDeleteModal = !this.showDeleteModal;
+  }
+
+  @action
+  handleDelete(deleteType) {
+    // deleteType should be 'delete', 'destroy', 'undelete', 'delete-latest-version', 'destroy-all-versions'
+    if (!deleteType) {
+      return;
+    }
+    if (deleteType === 'destroy-all-versions' || deleteType === 'v1') {
+      let { id } = this.args.model;
+      this.args.model.destroyRecord().then(() => {
+        this.args.navToNearestAncestor.perform(id);
+      });
+    } else {
+      return this.store
+        .adapterFor('secret-v2-version')
+        .v2DeleteOperation(this.store, this.args.modelForData.id, deleteType)
+        .then(resp => {
+          if (Ember.testing) {
+            return;
+          }
+          if (!resp) {
+            location.reload();
+          }
+          if (resp.isAdapterError) {
+            const errorMessage = getErrorMessage(resp.errors);
+            this.flashMessages.danger(errorMessage);
+          } else {
+            location.reload();
```

Comment:
here is the reload.  Curious if folks have other suggestions. I tried all sorts of things and nothing cleared the modal.